### DISCUSSION TITLE
Why can't I see graph of HR on Bike workout?

### DISCUSSION THREAD

**Discussion by jjjjr1** [philosowaffle/peloton-to-garmin#664] [Q&A]
New to this app...

I successfully uploaded my workouts to Garmin Connect using TCX format -
When I go to the workout in Garmin Connect, the app knows my Avg and Max heartrate for the workout, but I can't click through to see the graph of my HR over time.

I tried enabling "Include Time in HR Zones" under the advanced settings and resync but I didn't see any change.

I can see the HR data over time in the TCX file for the workout. Anyone know why I don't see the graph on Garmin Connect similar to when I record a workout directly from the watch to Garmin Connect or view the workout in Strava?

Thanks for all the functionality!

**Comment by philosowaffle** [comment]
No immediate ideas. However the easiest solution may be to switch to FIT files instead of TCX.  Overall you will see more data and have support for more features by using FIT files instead of TCX.

**Comment by mcgonago** [comment]
Edit your TCX file and rename TrackPoint to Trackpoint (lower case p) - and upload

You will see all the nice graphs (but no Training Effect - see other threads for that)

Also - not sure if 100% necessary - do not have a newline at end of TCX file

This appears to be a bug with P2G with respect to getting the Trackpoint info uploaded to Garmin.

**Comment by mcgonago** [comment]
To be a bit more specific.

After I sync my workout - I find the raw .tcx file in the saved p2g-webui/output/tcx directory.

I edit the .tcx file (renaming TrackPoint to Trackpoint) - then manually upload the .tcx file using Garmin connect on my laptop.

I have to delete the previous activity that P2G uploads when it syncs...
